src/cgmemmgr.cpp

```diff
@@ -462,11 +462,12 @@ class ROAllocator {
     virtual ~ROAllocator() {}
     virtual void finalize()
     {
-        if (exec) {
-            for (auto &alloc: allocations) {
-                sys::Memory::InvalidateInstructionCache(alloc.rt_addr,
-                                                        alloc.sz);
-            }
+        for (auto &alloc: allocations) {
+            // ensure the mapped pages are consistent
```

Comment:
Coherent is probably the right word since this is talking about the same memory location having different values simultaneously due to split caches.